services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/endpoints/UseGlobalEndpointResolverTest.java

```diff
@@ -118,6 +120,45 @@ public void differentCombinationOfConfigs_shouldResolveCorrectly() {
         assertThat(resolver.resolve(Region.US_EAST_1)).isEqualTo(testData.expected);
     }
 
+    @Test
+    public void useUsEast1RegionalEndpoint_fromProfileFileNotRegional_resolvesToTrue() {
```

Comment:
What happens if the profile file isn't set, will some test cases resolve to us-east-1 and some not?